dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java

```diff
@@ -1102,9 +1104,13 @@ private String recursionProcessDefinitionName(long projectCode, String processDe
             putMsg(result, Status.PROCESS_DEFINE_NOT_EXIST, codes);
             return result;
         }
+        HashMap<Long, Project> userProjects =  new HashMap(Constants.DEFAULT_HASH_MAP_SIZE);
```

Comment:
It's better to set the map size by db query result.

Reply:
> It's better to set the map size by db query result.

It's just a default initialize value, and the hash map size can be extended automatically.